src/common/telemetry.ts

```diff
@@ -1,377 +1,381 @@
+                            });
+            regKey.get(value, function(err: any, itemValue: winreg.RegistryItem) {
+                if (err) {
+                    // Fail gracefully by returning null if there was an error.
+                    deferred.resolve(null);
                 } else {
-                    return Q.resolve(fallback());
+                    deferred.resolve(itemValue.value);
                 }
-            }
+            });
 
-            private static generateMachineId(): Q.Promise<string> {
-                return TelemetryUtils.getMacAddress().then((macAddress: string) => {
-                    return crypto.createHash("sha256").update(macAddress, "utf8").digest("hex");
-                });
+            return deferred.promise;
+        }
+
+        /*
+            * Load settings data from settingsHome/TelemetrySettings.json
+            */
+        private static loadSettings(): ITelemetrySettings {
+            try {
+                TelemetryUtils.telemetrySettings = JSON.parse(<any>fs.readFileSync(TelemetryUtils.telemetrySettingsFile));
+            } catch (e) {
+                // if file does not exist or fails to parse then assume no settings are saved and start over
+                TelemetryUtils.telemetrySettings = {};
             }
 
-            private static getMachineId(): Q.Promise<string> {
-                let machineId: string = TelemetryUtils.telemetrySettings.machineId;
-                if (!machineId) {
-                    return TelemetryUtils.generateMachineId()
-                    .then(function(id: string): Q.Promise<string> {
-                        TelemetryUtils.telemetrySettings.machineId = id;
-                        return Q.resolve(id);
-                    });
-                } else {
-                    TelemetryUtils.telemetrySettings.machineId = machineId;
-                    return Q.resolve(machineId);
-                }
+            return TelemetryUtils.telemetrySettings;
+        }
+
+        /*
+            * Save settings data in settingsHome/TelemetrySettings.json
+            */
+        private static saveSettings(): void {
+            let settingsHome = HostPlatform.getSettingsHome();
+            if (!fs.existsSync(settingsHome)) {
+                fs.mkdirSync(settingsHome);
             }
 
-            private static getMacAddress(): Q.Promise<string> {
-                // Return a mac address, or failing that, a unique ID
-                // using getmac to attempt to match telemetry identifiers of vs code
-                return Q.nfcall(getmac.getMac).catch(() => {
-                    return TelemetryUtils.generateGuid();
+            fs.writeFileSync(TelemetryUtils.telemetrySettingsFile, JSON.stringify(TelemetryUtils.telemetrySettings));
+        }
+
+        private static getUniqueId(regValue: string, regHive: string, fallback: () => string): Q.Promise<any> {
+            let uniqueId: string;
+            if (os.platform() === "win32") {

```

Comment:
We can use `HostPlatform.getPlatformId() === HostPlatformId.WINDOWS` instead to avoid strings.